AllReadyApp/Web-App/AllReady/wwwroot/js/activity.js

```diff
@@ -1,57 +1,118 @@
+        self.errorUnenrolling = ko.observable(false);
 
-    var activityViewModel;
+        self.signupForActivity = function() {
+             HTBox.showModal({
+                 viewModel: new SignupViewModel(signupModel, self.unassociatedSkills),
+                 template: "VolunteerModal",
+                 context: self
+             }).then(activitySignupSuccess);
+        };
 
-    ActivityViewModel.prototype = {
-        enroll: function (activity) {
-            // TODO: set up a spinner
-            $.ajax({
-                type: "POST",
-                url: '/api/activity/' + activity + '/signup',
-                contentType: "application/json",
-            }).then(function (data) {
-                activityViewModel.enrolled(true);
-            }).fail(function (fail) {
-                console.log(fail);
+        function activitySignupSuccess(signUpViewModel) {
+            self.skills().forEach(function(skill) {
+                if (signUpViewModel.AddSkillIds().indexOf(skill.Id) >= 0){ skill.IsUserSkill(true) }
             });
-        },
-        unenroll: function (activity) {
-            activityViewModel.errorUnenrolling(false);
+            self.enrolled(true);
+            showalert("<strong>Thanks for volunteering! Your request has been processed and one of the activity coordinators will be in contact with you soon.</strong>", "alert-success", 30);
+        }
+
+        self.unenroll = function (activity) {
+            // TODO: set up a spinner
+            self.errorUnenrolling(false);
             $("#enrollUnenrollSpinner").show();
             $.ajax({
                 type: "DELETE",
                 url: '/api/activity/' + activity + '/signup',
-                contentType: "application/json",
+                contentType: "application/json"
             }).then(function (data) {
-                activityViewModel.enrolled(false);
+                self.enrolled(false);
                 $("#enrollUnenrollSpinner").hide();
+                showalert("<strong>Thanks for your interest. Your request has been processed and you are no longer signed up for this activity We hope to see you soon!.</strong>", "alert-success", 30);
             }).fail(function (fail) {
-                activityViewModel.errorUnenrolling(true);
-                $("#enrollUnenrollSpinner").hide();
+                self.errorUnenrolling(true);
                 console.log(fail);
             });
         }
+    }
+
+    function showalert(message, alerttype, timeoutInSecs) {
+        $('#alert_placeholder').append('<div id="alertdiv" class="alert ' +  alerttype + ' fade in"><a class="close" data-dismiss="alert">×</a><span>'+message+'</span></div>')
+        if (timeoutInSecs) {
+            setTimeout(function() {
+                $(".alert").alert('close');
+            }, timeoutInSecs * 1000);
+        };
+    };
+
+    SignupViewModel = function (signupModelSeed, unassociatedSkills) {

```

Comment:
This function will live in the global namespace unless you do one of these things:
* `var SignupViewModel = function (signupModelSeed, unassociatedSkills)`
*  `function SignupViewModel(signupModelSeed, unassociatedSkills)`